ddmrp/models/stock_buffer.py

```diff
@@ -743,12 +743,17 @@ def _get_product_sellers(self):
         if not sellers:
             # fallback to all sellers
             sellers = all_sellers
-        # When the current transaction changed the sequence, it may happen that
-        # the sellers' recordset is not correctly sorted by default.
-        sellers = sellers.sorted(key="sequence")
         return sellers
 
-    @api.depends("buffer_profile_id", "product_id.seller_ids")
+    @api.depends(
+        "buffer_profile_id",
+        "item_type",
+        "product_id.seller_ids",
+        "product_id.seller_ids.company_id",
+        "product_id.seller_ids.name",
+        "product_id.seller_ids.product_id",
+        "product_id.seller_ids.sequence",
```

Comment:
The main supplier is in standard based on "sequence, min_qty desc, price"
```suggestion
        "product_id.seller_ids.sequence",
        "product_id.seller_ids.min_qty",
        "product_id.seller_ids.price",
```